### ISSUE TITLE
PluginRequirements should not be fulfilled unless their PluginRequirements are fulfilled

### ISSUE THREAD

**Issue by ikelos** [volatilityfoundation/volatility3#1095]
A PluginRequirement should verify that the required plugins can successfully be instantiated.  As such we need to verify the subset of requirements that may be required (at the moment, only other plugin requirements and module requirements, possibly all version requirements).  It needs a little bit of thought, as to what's actually needed (most requirements are inputs, that aren't necessarily required for exposed methods), but at the moment a dependency may be fulfilled with a non-functional plugin.

**Comment by ikelos** [comment]
Not stale, still need to get this one sorted...

**Comment by ikelos** [comment]
Not stale!  I don't think it's been sorted yet either, no longer sure though, it was such a long time ago...?

**Comment by atcuno** [comment]
Did this get fixed in the parity push @ikelos ?

**Comment by ikelos** [comment]
Yep, I believe [this loop](https://github.com/volatilityfoundation/volatility3/blob/1a71b03b8da6203616951923e316a245ae396d05/volatility3/framework/configuration/requirements.py#L572) verifies that VersionRequirements validate all sub-version requirements...

**Comment by ikelos** [comment]
Sadly I didn't record (and possibly none exists) of the case I was envisaging, but I think it was around requirements somewhat becoming somewhat dual purpose.  Partially for running the plugin through run, and partially for class methods (which could depend on subplugins).  This almost makes two classes of requirements, execution and functional requirements I guess?  I think everything hangs together at the moment ok, but that's basically because we only have one type of functional requirement (the Version/PluginRequirements) and they get treated specially (by descending into them when doing a requirements check).  It might be worth separating these out at some point to make it more explicit.  I guess I missed it as a need when we started rolling classmethods out as a means of plugin functionality...
